src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -211,6 +332,83 @@ private IFeatureFilterMetadata GetFeatureFilterMetadata(string filterName)
             return filter;
         }
 
+        private IFeatureVariantAssignerMetadata GetFeatureAssignerMetadata(string assignerName)
+        {
+            const string assignerSuffix = "assigner";
+
+            IFeatureVariantAssignerMetadata assigner = _assignerMetadataCache.GetOrAdd(
+                assignerName,
+                (_) => {
+
+                    IEnumerable<IFeatureVariantAssignerMetadata> matchingAssigners = _variantAssigners.Where(a =>
+                    {
+                        Type filterType = a.GetType();
+
+                        string name = ((AssignerAliasAttribute)Attribute.GetCustomAttribute(filterType, typeof(AssignerAliasAttribute)))?.Alias;
+
+                        if (name == null)
+                        {
+                            name = GetTrimmedName(a.GetType(), assignerSuffix);
+                        }
+
+                        return IsMatchingMetadataName(name, assignerName);
```

Comment:
>IsMatchingMetadataName [](http://example.com/codeflow?start=31&length=22)

We probably can throw all the logic in the IsMatchingMetadataName including the alias attribute type, the trimming, and the suffix. Then the function can be shared between filter and assigner Get*Metadata.

Reply:
Actually, you can just put those 3 lines (356-378) in your `IsMatchingMetadataName` with an extra parameter of "suffix". Then you don't have to repeat the same logic for `GetFeatureVariantAssignerMetadata` and `GetFeatureFilterMetadata`.

Reply:
I'm not sure what iteration you're referring to with line numbers 356-378 to see what section you mean. I was unable to combine add any more code into IsMatchingMetadataName without it looking like overstretching to me. Even with suffix parameter.

Reply:
Sorry, it's hard to tell indeed :)

What I mean is we can replace this

```cs
return assignerName.EndsWith(assignerSuffix, StringComparison.OrdinalIgnoreCase) ?
    IsMatchingMetadataName(name, assignerName) :
    IsMatchingMetadataName(GetTrimmedName(name, assignerSuffix), assignerName);
```

with

```cs
return IsMatchingMetadataName(name, assignerName, suffix);
```

The logic you have in the current "return" can go into the new IsMatchingMetadataName with the extra `suffix` parameter.

```cs
private static bool IsMatchingMetadataName(string metadataName, string desiredName, string suffix)
```

If we do it, we can replace the similar lines in `GetFeatureFilterMetadata`.

Reply:
Oh okay, I added those in the last commit which was after the original comment that's why I was a bit confused. It's clear now though, I'll make that change.

Reply:
Updated.